src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,233 @@
+                      * (density_arg * std::exp(-log_r2r3)
+                         + 27 * density_arg * std::exp(-log_r2r3));
+
+    // Outgoing secondaries are electron and positron
+    electron_pair[0].def_id = shared_.electron_id;
+    electron_pair[1].def_id = shared_.positron_id;
+    // Energies of secondaries shared equally
+    electron_pair[0].energy
+        = units::MevEnergy{0.5 * epsilon * inc_energy_.value()};
+    electron_pair[1].energy = electron_pair[0].energy;
+
+    // Azimuthal angle sampled isotropically
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    electron_pair[0].direction = rotate(
+        from_spherical(std::cos(theta), sample_phi(rng)), inc_direction_);
+    // Opposite direction in costheta
+    electron_pair[1].direction = rotate(
+        from_spherical(-std::cos(theta), sample_phi(rng)), inc_direction_);
+    // Calculate produced pair directions via conservation of momentum
+    for (unsigned int p = 0; p < 2; p++)
+    {
+        // Loop over directions
+        for (int i = 0; i < 3; ++i)
+        {
+            electron_pair[p].direction[i]
+                = inc_direction_[i] * inc_energy_.value()
+                  - electron_pair[p].direction[i]
+                        * electron_pair[p].energy.value();
+        }
+        normalize_direction(&electron_pair[p].direction);
+    }
+
+    return result;
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::impact_parameter(size_type Z,
+                                                       real_type eps) const
+{
+    return 136 * std::cbrt(1 / Z) * epsilon0_ * epsilon0_
+           / (eps * (1 - eps));
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::screening_phi1(real_type delta) const
+{
+    if (delta <= 1)
+    {
+        return 20.867 - 3.242 * delta + 0.625 * delta * delta;
+    }
+    return screening_phi12(delta);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::screening_phi2(real_type delta) const
+{
+    if (delta <= 1)
+    {
+        return 20.209 - 1.930 * delta - 0.086 * delta * delta;
+    }
+    return screening_phi12(delta);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::screening_phi12(real_type delta) const
+{
+    return 21.12 - 4.184 * std::log(delta + 0.952);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::CoulombCorr(size_type Z) const
+{
+    real_type term1 = 8.0 / 3.0 * std::log(Z);
+
+    if (inc_energy_.value() < 50.0) // 50 MeV
+    {
+        return term1;
+    }
+
+    return term1 + 8 * this->CoulombCorr_aux(Z);
+}
+
+CELER_FUNCTION real_type BetheHeitlerInteractor::CoulombCorr_aux(size_type Z) const
+{
+    real_type alphaZ_sq = constants::alpha_fine_structure
+                          * constants::alpha_fine_structure * Z * Z;
+    real_type alphaZ_sqsq = alphaZ_sq * alphaZ_sq;
+    return alphaZ_sq
+           * (1 / (1 + alphaZ_sq) + 0.20206 - 0.0369 * alphaZ_sq
+              + 0.0083 * alphaZ_sqsq - 0.0020 * alphaZ_sq * alphaZ_sqsq);
```

Comment:
I use this formula to calculate the Tsai radiation length in `material/detail/Utils.cc`, so since it only depends on Z, we should probably precalculate it and store it on the `ElementDef`. Want me to do that in a separate merge request?

Reply:
Works for me!

Reply:
If your branch is up-to-date with master, you should be able to replace this whole function with `return element_.coulomb_correction()` (or just replace in the functions that call it).